lang/test/org/partiql/lang/errors/ParserErrorsTest.kt

```diff
@@ -1246,6 +1246,116 @@ class ParserErrorsTest : TestBase() {
+            Property.TOKEN_VALUE to ion.newSymbol("set")))
+
+    @Test
+    fun updateWithRemove() = checkInputThrowingParserException(
+        "UPDATE test SET x = REMOVE y",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 21L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("remove")))
+
+    @Test
+    fun updateWithInsert() = checkInputThrowingParserException(
+        "UPDATE test SET x = INSERT INTO foo VALUE 1",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 21L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("insert_into")))
+
+    @Test
+    fun updateWithDelete() = checkInputThrowingParserException(
+        "UPDATE test SET x = DELETE FROM y",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 21L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("delete")))
+
+    @Test
+    fun updateWithExec() = checkInputThrowingParserException(
+        "UPDATE test SET x = EXEC foo arg1, arg2",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 26L,
+            Property.TOKEN_TYPE to TokenType.IDENTIFIER,
+            Property.TOKEN_VALUE to ion.newSymbol("foo")))
+
+    @Test
+    fun updateWithCreateTable() = checkInputThrowingParserException(
+        "UPDATE test SET x = CREATE TABLE foo",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 21L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("create")))
+
+    @Test
+    fun updateWithDropTable() = checkInputThrowingParserException(
+        "UPDATE test SET x = DROP TABLE foo",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 21L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("drop")))
+
+    @Test
+    fun updateWithCreateIndex() = checkInputThrowingParserException(
+        "UPDATE test SET x = CREATE INDEX ON foo (x, y.z)",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 21L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("create")))
+
+    @Test
+    fun nestedRemove() = checkInputThrowingParserException(
+        "REMOVE REMOVE y",
+        ErrorCode.PARSE_INVALID_PATH_COMPONENT,
```

Comment:
Is there a reason this error is not `PARSE_UNEXPECTED_TERM`?

Reply:
This is handled actively by the parser while parsing the `REMOVE` keyword.